File: solutions/zig/collatz-conjecture/1/collatz_conjecture.zig
// Please implement the `ComputationError.IllegalArgument` error.

pub const ComputationError = error{IllegalArgument};

pub fn steps(number: usize) anyerror!usize {
    if (number < 1) {
        return ComputationError.IllegalArgument;
    } else if (number == 1) {
        return 0;
    } else if (number % 2 == 0) {
        const prev = try steps(number / 2);
        return 1 + prev;
    } else {
        const prev = try steps(number * 3 + 1);
        return 1 + prev;
    }
}

File: solutions/csharp/collatz-conjecture/1/CollatzConjecture.cs
public static class CollatzConjecture
{
    public static int Steps(int number)
    {
        if (number < 1)
        {
            throw new ArgumentOutOfRangeException();
        }
        var n = number;
        var ans = 0;
        while (n != 1)
        {
            ans += 1;
            if (n % 2 == 0)
            {
                n /= 2;
            } else
            {
                n = 3 * n + 1;
            }
        }
        return ans;
    }
}